design/src/rfcs/rfc0006_crates_io_publishing.md

```diff
@@ -0,0 +1,279 @@
+and it turns out there's a problem that requires yanking one of them, then all 260 will be yanked. Attempting to do
+partial yanking will require a lot of effort and be difficult to get right. Yanking should be a last resort.
+
+Concrete Scenarios
+------------------
+
+The following are example scenarios that may occur that the publishing implementation must handle.
+
+1. **AWS models were updated**: This results in `patch` version bumps for all AWS SDK crates _that had changes_.
+2. **Bug fixed in runtime crate**: Runtime crate is `minor` version bumped. All runtime crates that depend on it are
+   also `minor` version bumped, and all of the AWS SDK crates are `minor` version bumped.
+3. **Bug fixed in codegen**: All AWS SDK crates are `minor` version bumped.
+4. **CVE discovered in external dependency**: All manifests depending on the vulnerable dependency are
+   updated to explicitly depend on the fixed version (or newer). If these changes are isolated to the runtime crates,
+   then `patch` revisions of those crates can be published. Otherwise, this will result in a `minor` version bump
+   in runtime crates and AWS SDK crates.
+5. **CVE discovered in runtime crate**: If it can be fixed in a backwards compatible way, then `patch` revision the
+   runtime crate with the fix. Otherwise, `minor` version bump the runtime crate and follow the normal process from there.
+6. **CVE discovered in generated code**: Same process as a bug fix in codegen.
+7. **Accidental release of breaking change**: Run publish process in reverse, yanking the published packages.
+8. **Buggy update to runtime crate**: If possible, roll forward with a `patch` revision. Otherwise, run publish process
+   in reverse yanking the published packages.
+
+Short-term Proposal
+-------------------
+
+The short-term approach builds off our current weekly release process.
+That process, at time of writing, is the following:
+
+1. Run script to update AWS models
+2. Manually update AWS SDK version in `aws/sdk/gradle.properties` in smithy-rs
+3. Tag smithy-rs
+4. Wait for GitHub actions to generate AWS SDK using newly released smithy-rs
+5. Check out aws-sdk-rust, delete existing SDK code, unzip generated SDK in place, and update readme
+6. Tag aws-sdk-rust
+
+To keep things simple:
+- The Smithy runtime crates will have the same smithy-rs version
+- All AWS crates will have the same AWS SDK version
+- There will _NOT_ be `patch` revisions
+
+All runtime crate version numbers in smithy-rs will be locked at `0.0.0-smithy-rs-head`. This is a fake
+version number that gets replaced when generating the SDK.
+
+The SDK generator script in smithy-rs will be updated to:
+- Replace Smithy runtime crate versions with the smithy-rs version from `aws/sdk/gradle.properties`
+- Replace AWS runtime crate versions with AWS SDK version from `aws/sdk/gradle.properties`
+- Add correct version numbers to all path dependencies in all the final crates that end up in the build artifacts
```

Comment:
Now that I'm implementing this, I think I'm going to move this last step into the publish script since that will be easier. We need to do these kinds of updates in the long-term solution anyway, so that should be future proof.